docker/Dockerfile.dmqnode

```diff
@@ -1,26 +1,62 @@
 # Duplicate definition of DIST before FROM is needed to be able to use it
 # in docker image name:
 ARG  DIST=xenial
-FROM sociomantictsunami/develdlang:$DIST-v6 as builder
+FROM sociomantictsunami/develdlang:$DIST-v7 as builder
 # Copies the whole project as makd needs git history:
 COPY . /project/
 WORKDIR /project/
 # Redefine arguments as env vars to be used inside build.sh script:
 ARG DIST=xenial
 ARG DMD=2.078.*
-ENV DMD=$DMD DIST=$DIST
-RUN docker/build.sh
+ENV DMD=$DMD
+RUN set -eu \
+    # Compiler configuration \
+    && . submodules/beaver/lib/dlang.sh \
+    \
+    && set_dc_dver \
+    \
+    # Install dependencies \
+    \
+    && case "$DMD" in \
+        dmd*   ) PKG= ;; \
+        1.*    ) PKG="dmd1=$DMD-$DIST" ;; \
+        2.*.s* ) PKG="dmd-transitional=$DMD-$DIST" ;; \
+        2.*    ) PKG="dmd-compiler=$DMD dmd-tools=$DMD libphobos2-dev=$DMD" ;; \
+        *      ) echo "Unknown \$DMD ($DMD)" >&2; exit 1 ;; \
+    esac \
+    \
+    && apt update \
+    && apt install -y --allow-downgrades \
+        $PKG \
+        libebtree6-dev \
+        liblzo2-dev \
+        libglib2.0-dev \
+        libpcre3-dev \
+        libgcrypt-dev \
+        libgpg-error-dev \
+    \
+    # Build app \
+    && if_d 2 make d2conv \
+    && make all pkg F=production
```

Comment:
Please put this in a separate script, this is awful to maintain and very error prone. At the level of the Dockerfile the only interesting information is that you are building the project[1], all the small details you generally don't care about, the same as you do `make all` instead of writing every single command in the Dockerfile.

[1] Maybe the dependencies too, but you fix that with just leaving the `apt install` in the dockerfile for everything that is not the compiler. If you also want to keep the `F=production` explicit, just do a `RUN F=production ./build`.